src/Coordination/KeeperStorage.h

```diff
@@ -35,40 +33,129 @@ class KeeperStorage
     /// New fields should be added to the struct only if it's really necessary
     struct Node
     {
-        /// to reduce size of the Node struct we use a custom Stat without dataLength
-        struct Stat
+        int64_t czxid{0};
+        int64_t mzxid{0};
+        int64_t pzxid{0};
+        uint64_t acl_id = 0; /// 0 -- no ACL by default
+
+        int64_t mtime{0};
+
+        struct
         {
-            int64_t czxid{0};
-            int64_t mzxid{0};
-            int64_t ctime{0};
-            int64_t mtime{0};
-            int32_t version{0};
-            int32_t cversion{0};
-            int32_t aversion{0};
-            int32_t numChildren{0}; /// NOLINT
-            int64_t ephemeralOwner{0}; /// NOLINT
-            int64_t pzxid{0};
-
-            bool operator==(const Stat &) const = default;
-        };
+            bool is_ephemeral : 1;
+            int64_t ctime : 63;
```

Comment:
Please leave a comment about the consequences and why it's safe. It will be super useful to whoever reads this later on